python/ray/serve/controller.py

```diff
@@ -364,8 +364,13 @@ def deploy(
 
         autoscaling_config = deployment_config.autoscaling_config
         if autoscaling_config is not None:
-            # TODO: is this the desired behaviour? Should this be a setting?
-            deployment_config.num_replicas = autoscaling_config.min_replicas
+            previous_deployment = self.deployment_state_manager.get_deployment(name)
```

Comment:
[Non blocking merge idea] Is it possible we directly make all these logics (L366 - L373) into the `_set_target_state` of DeploymentState class?

Reply:
no because the deployment state is not aware of historical deployments (or previous version at the spec)

Reply:
We have https://github.com/ray-project/ray/blob/master/python/ray/serve/_private/deployment_state.py#L1993, if the deployment exists, you will be in existing deployment to update your config, so you should have all information.

My two cents:

- only have one place to update the num_replicas which is better inside the DeploymentState.
- Your current change is under the same structure as before, no need to change in this pr :)